### PR TITLE
framework/st_things : Change strdup to things_strdup

### PR DESCRIPTION
**Pull request by JeongJunSik**
- Change to things_strdup for debugging by st_things module

### CODE REVIEW

File: framework/src/st_things/things_stack/memory/things_string.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        return NULL;
+    }
+
+    // Allocate memory for original string length and 1 extra byte for '\0'
+    size_t length = strlen(str);
+    char *dup = (char *)things_malloc(length + 1);
+    if (NULL != dup) {
+        memcpy(dup, str, length + 1);
+    }
+
```
Comment (girishks2000) [inline framework/src/st_things/things_stack/memory/things_string.c]: Dont you think copying actual data of length size and you appending '\0'  at the end is more readable and conveys the purpose

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/memory/things_string.c:29: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:30: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/memory/things_string.c:30: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:31: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:33: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:34: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:35: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:36: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:37: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/memory/things_string.c:37: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:38: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:40: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: framework/src/st_things/things_stack/fota/fmwup_util_data.c [hunk 2]
```diff
@@ -142,17 +142,17 @@ char *key_manager_set_default_data(const char *name)
 	char *default_data = NULL;
 
 	if (strcmp(name, FIRMWARE_PROPERTY_UPDATE) == 0) {
-		default_data = strdup(KEY_MANAGER_INT_DEFAULT_DATA);
+		default_data = things_things_strdup(KEY_MANAGER_INT_DEFAULT_DATA);
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/fota/fmwup_util_data.c]: Modify ```things_things_```

File: framework/src/st_things/things_stack/memory/things_string.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/memory/things_string.c]: 2018

File: framework/src/st_things/things_stack/memory/things_string.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+#include <string.h>
+#include <assert.h>
+#include "things_malloc.h"
+
+#define TAG "THINGS_STRING"
+
+char *things_strdup(const char *str)
+{
+    if (!str) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/memory/things_string.c]: check violations of coding rules in this file

File: framework/src/st_things/things_stack/memory/things_string.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/******************************************************************
+ *
+ * Copyright 2014 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/memory/things_string.h]: 2018 and remove unnecessary empty line

File: framework/src/st_things/things_stack/memory/things_string.c [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ ****************************************************************************/
+
+#include "things_string.h"
+
+#include <string.h>
+#include <assert.h>
+#include "things_malloc.h"
+
+#define TAG "THINGS_STRING"
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/memory/things_string.c]: What's the purpose of this definition?

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/memory/things_string.c:27: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:28: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/memory/things_string.c:28: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:29: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:32: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:33: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:34: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:35: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/memory/things_string.c:35: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:36: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/memory/things_string.c:38: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**